typing/types.mli

```diff
@@ -331,6 +333,12 @@ and constructor_arguments =
   | Cstr_tuple of type_expr list
   | Cstr_record of label_declaration list
 
+and unboxed_status =
```

Comment:
>I don't think there would be any significant memory overhead with a record since the 4 possible values will be stored as structured constants (and thus shared). Are you concerned with the extra memory load?

Oh, I didn't think of that. It's so nice I think I'll change the code. I don't think the extra access time is worse than calling the helper functions.